app/Http/Requests/UpdateUserRequest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpdateUserRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        if($this->password != ''){
+            return [
+                'full_name' => 'required|regex:/^[\pL\s\-]+$/u',
+                'email' => 'required|email|unique:users,email,'.$this->user->id,
+                'address' => 'required',
+                'phone_number' => 'required|regex:/[0-9]{10}/',
+                'password' => 'min:6|max:20',
```

Comment:
form update password nên có 2 input (để trường hợp ng dùng nhập bị sai do bàn phím,nếu ko có confirm thì lúc đổi pass xong lại ko đăng nhập đc ),password, password_confirmation => ở đây password phải thêm rule confirmed

Reply:
'password' => 'min:6|max:20|confirmed'

Reply:
$rule = [
                'full_name' => 'required|regex:/^[\pL\s\-]+$/u',
                'email' => 'required|email|unique:users,email,'.$this->user->id,
                'address' => 'required',
                'phone_number' => 'required|regex:/[0-9]{10}/',
            ];
if (!empty($this->password)) {
    $rule['password']= 'min:6|max:20|confirmed';
}
return $rule;